inc/composer/class-command.php

```diff
@@ -654,6 +661,179 @@ protected function db( InputInterface $input, OutputInterface $output ) {
+						break;
+					# Linux
+					case 'Linux':
+						$binary_arch = $arch === 'amd64' ? 'linux-amd64' : 'linux-arm64';
+						break;
+					# Windows
+					case 'Windows':
+						$binary_arch = 'windows-amd64.exe';
+						break;
+					default:
+						$binary_arch = null;
+						break;
+				}
+
+				// If couldn't detect a support architecture, ask the user to install mkcert manually.
+				if ( ! $binary_arch ) {
+					$output->writeln( '<error>This command is only supported on macOS, Linux, and Windows x64, install `mkcert` manually for other systems.</error>' );
+					return 1;
+				}
+
+				$binary = "mkcert-$mkcert_version-$binary_arch";
+				$mkcert = "vendor/mkcert";
+
+				// Check if mkcert is installed globally already, bail if so.
+				$version = trim( shell_exec( 'mkcert -version' ) );
+				if ( $version ) {
+					$output->writeln( "<error>mkcert $version is installed globally already</error>" );
+					return 1;
+				}
+
+				// Check if mkcert is installed locally already, bail if so.
+				$version = trim( shell_exec( "$mkcert -version" ) );
+				if ( $version ) {
+					$output->writeln( "<error>mkcert $version is installed locally already</error>" );
+					return 1;
+				}
+
+				exec( "curl -o $mkcert -L https://github.com/FiloSottile/mkcert/releases/download/$mkcert_version/$binary", $dummy, $result );
+				if ( $result ) {
+					$output->writeln( "<error>Could not download mkcert binary, try using sudo or manually installing mkcert.</error>" );
+					return 1;
+				}
+
+				$output->writeln( "<info>mkcert $mkcert_version was downloaded.</info>" );
+
+				chmod( $mkcert, 0755);
+
+				exec( "$mkcert -version", $dummy, $result );
+				if ( $result ) {
+					$output->writeln( "<error>Could not launch mkcert binary, try manually installing mkcert.</error>" );
+					return 1;
+				}
+				$output->writeln( "<info>mkcert $mkcert_version was installed.</info>" );
+
+				// Setup and accept the root certificate.
+				exec( "$mkcert -install", $dummy, $result );
+				if ( $result ) {
+					$output->writeln( "<error>Could not setup mkcert properly, try manually installing mkcert.</error>" );
+					return 1;
+				}
+
+				$output->writeln( "<info>mkcert root CA was installed and accepted successfully.</info>" );
+				return 0;
+				break;
+			case 'generate':
+				$mkcert = $this->get_mkcert_binary();
+				if ( ! $mkcert ) {
+					$output->writeln( "<error>mkcert is not installed, run 'composer server ssl install' or install mkcert manually first.</error>" );
+					return 1;
+				}
+
+				// TODO figure out how to programmatically detect the domains to use
+				$domains = $input->getArgument( 'options' )[1] ?? '*.altis.dev';
+
+				exec( "$mkcert -cert-file vendor/ssl-cert.pem -key-file vendor/ssl-key.pem $domains", $dummy, $result );
+
+				if ( $result ) {
+					$output->writeln( "<error>Could not generate certificates! Try generating them manually using mkcert.</error>" );
```

Comment:
Lets provide an example command so users know the intended cert file and key file locations. Also will it generate these files per domain or all in one file? Just wondering about overwriting. Likely not a problem but something to consider.

Reply:
It should do it for all domains, however, we don't have a way to specify _multiple_ domains just yet, but when we do, the command can be configured to do that. Traefik is configured to use that one certificate as the default one.